gunrock/cuda/launch_box.hxx

```diff
@@ -0,0 +1,131 @@
+/**
+ * @file device_properties.hxx
+ * @author Cameron Shinn (ctshinn@ucdavis.edu)
+ * @brief
+ * @version 0.1
+ * @date 2020-11-09
+ *
+ * @copyright Copyright (c) 2020
+ *
+ */
+#pragma once
+
+#include <type_traits>
+
+namespace gunrock {
+namespace cuda {
+
+/**
+ * @brief Struct holding kernel parameters will be passed in upon launch
+ * @tparam block_dimensions_ Block dimensions to launch with
+ * @tparam grid_dimensions_ Grid dimensions to launch with
+ * @tparam smem_bytes_ Amount of shared memory to allocate
+ *
+ * @todo dimensions should be dim3 instead of unsigned int
+ */
+template<
+  unsigned int block_dimensions_,
+  unsigned int grid_dimensions_,
+  unsigned int smem_bytes_ = 0
+>
+struct launch_params_t {
+  enum : unsigned int {
+    block_dimensions = block_dimensions_,
+    grid_dimensions = grid_dimensions_,
+    smem_bytes = smem_bytes_
+  };
+};
+
+#define TEST_SM 75  // Temporary until we figure out how to get cabability combined
```

Comment:
You're looking for a concatenate macro here, lots of examples for this.

Reply:
Misworded my comment here. I meant just like getting the entire value on the host-side at compile time. I saw your latest comment.